File: backend/src/main/java/com/abel/gastrohub/restaurant/Restaurant.java
package com.abel.gastrohub.restaurant;

import com.abel.gastrohub.user.User;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

import java.time.LocalDateTime;
import java.util.LinkedHashSet;
import java.util.Set;

@Setter
@Getter
@Entity
@Table(name = "restaurants")
public class Restaurant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 255)
    @NotNull
    @Column(name = "name", nullable = false)
    private String name;

    @Size(max = 255)
    @NotNull
    @Column(name = "address", nullable = false)
    private String address;

    @Size(max = 100)
    @NotNull
    @Column(name = "cuisine_type", nullable = false, length = 100)
    private String cuisineType;

    @NotNull
    @Column(name = "description", nullable = false, length = Integer.MAX_VALUE)
    private String description;

    @NotNull
    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @NotNull
    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @OneToMany(mappedBy = "restaurant")
    private Set<User> users = new LinkedHashSet<>();

    @Size(max = 20)
    @Column(name = "invitation_code", length = 20)
    private String invitationCode;

    @Column(name = "invitation_expires_at")
    private LocalDateTime invitationExpiresAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }

}

File: frontend/lib/src/features/restaurant/models/restaurant.dart
class Restaurant {
  final int id;
  final String name;
  final String address;
  final String cuisineType;
  final String description;
  final int paymentPlanId;

  Restaurant({
    required this.id,
    required this.name,
    required this.address,
    required this.cuisineType,
    required this.description,
    required this.paymentPlanId,
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['id'] as int,
      name: json['name'] as String,
      address: json['address'] as String,
      cuisineType: json['cuisineType'] as String,
      description: json['description'] as String,
      paymentPlanId: json['paymentPlanId'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'address': address,
      'cuisineType': cuisineType,
      'description': description,
      'paymentPlanId': paymentPlanId,
    };
  }
}
